src/urbit/app/chess.hoon

```diff
@@ -52,7 +63,8 @@
   ^-  (quip card _this)
   =/  old-state  !<(versioned-state old-state-vase)
   ?-  -.old-state
-    %0  [~ this(state old-state)]
+    %1  [~ this(state old-state)]
+    %0  [~ this(state 1+old-state)]
```

Comment:
`on-load` defines logic for how to revive an app that has been suspended. It's the counterpart to `on-save`: `on-save` defines how to convert the state of the app to a noun before going offline, and `on-load` defines how to convert a noun to application state before coming back online.

Since the process for going offline and then coming back online is also what an application does between version updates, `on-load` also defines state transition logic when the form of the application state changes between versions.

That means that when a version goes offline with state version `%0` and comes back to an application where state version `%1` is the latest state, it needs to have logic for converting its state from version `%0` to version `%`.

I recommend you re-read [chapter 4 of App School](https://developers.urbit.org/guides/core/app-school/4-lifecycle) to get a better grasp of the concept.

I'm going to quote my earlier thoughts from our conversation on Urbit (and add @bonbud-macryg & @brbenji 
 to the conversation below since it's relevant to all of you):
> Since `active-game-state` is a part of the application state (it's a convenience-type that's declared to make the `state-0` type easier to specify), any changes to it are indirect changes to `state-0`, and therefore it needs to be versioned to `state-1`. But the whole point of versioned states is for applications to be able to cleanly upgrade state using the `on-load` arm, once they're revived after an update. Therefore, it's possible that we may need to not only incrementally version `active-game-state`, but to also keep another copy of it which retains the spec it had in state-0. This also brings up an interesting question: what if we needed to do this for a type declared in `../sur/chess.hoon` (e.g. `chess-game`)?
>
> I see two designs, and which one we choose to employ depends on the specific changes: more specifically, it depends on what would make the state transition logic in `on-load` easier to implement, which is downstream of the specific changes.
>
> Design 1:
> - Modify the lesser type (e.g. `active-game-state`), do not incrementally version it
> - Bump app state version
> - In `on-load` arm, treat old instances of the lesser type as raw cells, access fields directly
  (i.e. `+6.old-game-state` instead of `position.old-game-state`)
> 
> Design 2:
> - Create a file for outdated types: `../sur/historic.hoon`
> - Copy the current version of the lesser type (e.g. `active-game-state`) to `../sur/historic.hoon`
  and append the state version at the time to its name (e.g. `active-game-state-0` for the version
  of `active-game-state` for `state-0`)
> - Modify the lesser type, keep its name as is
> - Bump app state version
> - Import `../sur/historic.hoon` into `../app/chess.hoon`
> - In `on-load` arm, refer to old instances of the lesser type using the historic types

I've been thinking about this lately, and I think that these are good designs, but that the combined changes for all of Milestone 1 will be so large and fix so many bugs that make `%chess` unpleasant to deal with currently, that we should just reset the state completely for the release. **Therefore, don't worry about implementing state transition logic: if you need to modify the state, just make `state-1` completely reset the entire app state.**